pkg/ottl/expression_test.go

```diff
@@ -266,6 +306,86 @@ func Test_newGetter(t *testing.T) {
 			},
 			want: "pass",
 		},
+		{
+			name: "function call nested SliceBool",
+			val: value{
+				Literal: &mathExprLiteral{
+					Converter: &converter{
+						Function: "SliceBool",
+						Keys: []key{
+							{
+								// note for review: not sure if this is correct
```

Comment:
I think this is fine as-is. We're indexing the outer `[]any` slice, then the inner `[]bool` slice, so we have two integer keys here.

Reply:
ohh okay I was struggling to understand what this was for. Not sure if there's a better option than this though -- the joys of abstraction